pkg/components/node-problem-detector/component.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package nodeproblemdetector is code for node-problem-detector.
+package nodeproblemdetector
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+
+	"github.com/kinvolk/lokomotive/internal/template"
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
+)
+
+const (
+	// Name represents node-problem-detector component name as it should be referenced in function calls
+	// and in configuration.
+	Name = "node-problem-detector"
+)
+
+type component struct {
+	Namespace         string
+	CustomMonitors    []string `hcl:"custom_monitors,optional"`
+	CustomMonitorsRaw map[string]string
+}
+
+// NewConfig returns new node-problem-detector component configuration with default values set.
+//
+//nolint:golint
+func NewConfig() *component {
+	return &component{
+		Namespace: "kube-system",
+	}
+}
+
+// LoadConfig loads the component config.
+func (c *component) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContext) hcl.Diagnostics {
+	if configBody == nil {
+		return hcl.Diagnostics{
+			components.HCLDiagConfigBodyNil,
+		}
+	}
+
+	if err := gohcl.DecodeBody(*configBody, evalContext, c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// RenderManifests renders the Helm chart templates with values provided.
+func (c *component) RenderManifests() (map[string]string, error) {
+	helmChart, err := components.Chart(Name)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving chart from assets: %w", err)
+	}
+
+	customMonitorsMap := map[string]string{}
+
+	for key, val := range c.CustomMonitors {
+		fileName := "custom-monitor-" + strconv.Itoa(key) + ".json"
+		customMonitorsMap[fileName] = val
+	}
+
+	c.CustomMonitorsRaw = customMonitorsMap
```

Comment:
Can't we do this with loop in template? Or take a map from the user?

Reply:
Done it as a loop in template